server/testkit/tests/proto_v1_test.rs

```diff
@@ -1460,3 +1462,178 @@ async fn test_server_user_auth_reauthentication(rsclient: KanidmClient) {
     eprintln!("{:?} {:?}", now, uat.purpose);
     assert!(uat.purpose_readwrite_active(now));
 }
+
+macro_rules! start_password_session {
```

Comment:
Generally we should avoid macros unless really needed for small ergonomic tasks. Better to use functions.